File: pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import javax.naming.Name;
import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String name);

    void deleAddr(Long id);

    List<Provinces> findAllAddress();

    List<Cities> findCityById(String provinceid);

    List<Areas> findAreaById(String cityid);

    void save(Address entity,String name);
}

File: pinyougou-web-user/src/main/webapp/js/service/addressService.js
//购物车服务层
app.service('addressService',function($http){

    //获取当前登录账号的收货地址
    this.findAddressList=function(){
        return $http.get('address/findListByLoginUser.do');
    }

    //用户删除地址
    this.deleAddr=function(id){
        return $http.get('address/deleAddr.do?id='+ id);
    }

    //省市联动
    this.findAllAddress=function () {
        return $http.get('address/findAllAddress.do')
    }
    //查找市
    this.findCityById=function (provinceid) {
        return $http.get('address/findCityById.do?provinceid='+provinceid)
    }

    //查找县
    this.findAreaById=function (cityid) {
        return $http.get('address/findAreaById.do?cityid='+cityid)
    }

    //新增地址
    this.save = function (entity) {
        return $http.post('address/add.do',entity)
    }
});